apps/server/src/modules/task-card/uc/task-card.uc.ts

```diff
@@ -29,20 +20,13 @@ export class TaskCardUc {
 
 	async create(userId: EntityId, params: ITaskCardCRUD) {
 		const user = await this.authorizationService.getUserWithPermissions(userId);
-		let course: Course | undefined;
+		const course = await this.courseRepo.findById(params.courseId);
+		this.authorizationService.checkPermission(user, course, PermissionContextBuilder.write([]));
```

Comment:
- I vote for doing the `hasAllPermissions` check before the course stuff
- specific reason why to implement the `checkPermission` differently (and without throw) then the user check?

Reply:
I changed the order. I can't say anything to your second remark I just extended `checkPermission` that course is involved.

Reply:
I would like @virgilchiriac to have a look on that topic as I am not sure how to handle that properly.

Reply:
resolved, `checkPermission` throws by itself, so no need to separate that.